src/org/zaproxy/zap/extension/spider/SpiderDialog.java

```diff
@@ -112,8 +112,9 @@ public void init(Target target) {
         this.removeAllFields();
 
         this.addTargetSelectField(0, FIELD_START, this.target, true, false);
-        this.addComboField(0, FIELD_CONTEXT, new String[] {}, "");
+        this.addComboField(0, FIELD_CONTEXT, getContextNames(), "");
```

Comment:
If we do this always (instead of just when there's no `target`?) we should validate that the starting point is not in conflict with selected context (e.g. as is now it's possible to selected "Context X" while having as starting point a "Context Y").

Reply:
I'll have a look at it, maybe I've over simplified something that was done intentionally.

Reply:
Maybe, but this change makes the selection of the context/user more straightforward. If we don't pre-populate the context combo box when there's already a `target` we don't need to validate.

Reply:
Ok maybe I just need to re-work this slightly. There is still a consistency issue with the original behavior. Though not the consistency issue I was initially aiming to correct (I was trying to make the button functionality closer to the context menu functionality.).

1. Open a session that has a context defined.
2. Don't select anything in the sites/context tree.
3. Click "New Scan" in the spider tab, "Select" a starting point within your context.
4. Hit cancel.
5. Hit "New Scan" again. 
6. Note: The starting point is still set, but the context selection is empty.

So I guess when a `SpiderDialog` is opened we should check if there is a target (starting point) and use that to set the context and user combo fields?
(Or is it the cancel/clear functionality that should be resetting the target?)

Reply:
Both? We should use the existing target to set up the combo boxes and cancelling the dialogue should restore the previous settings (iirc, we talked about that already).

Reply:
I think we talked about it for active scan policies, and elsewhere probably.... or I may have forgotten (I apologize if it's the later).